src/macros.rs

```diff
@@ -0,0 +1,101 @@
+/// Emitt a tracing error event for an error with rich, structured context.
+/// It capturing the error using the experimental Rust [std::error::Report](https://doc.rust-lang.org/stable/std/error/struct.Report.html)
+/// and adding the type name as error.kind, the backtrace as error.trace and the error stack as error.message
+///
+/// # Examples
+///
+/// ```rust
+/// use prima_tracing::trace_error;
+/// # fn main() {
+///
+/// let error = "not a number".parse::<usize>().unwrap_err();
+/// trace_error!(error, "Parsing error!");
+/// trace_error!(error, "Parsing error: {}", uid="1234");
+/// # }
+/// ```
+#[macro_export]
+macro_rules! trace_error {
+    ($error:expr, $message:expr) => {
+        {
+            $crate::trace_error!($error, $message,)
+        }
+    };
+    ($error:expr, $message:expr, $($field:tt)*) => {
+        {
+            let kind = std::any::type_name_of_val(&$error);
+            let error_message = format!("{:#}", $error);
+            let stack = error_report::Report::new(&$error);
+            let trace = std::backtrace::Backtrace::force_capture();
+
+            $crate::tracing::error!(
+                {
+                    error.message = error_message,
+                    error.kind = kind,
+                    error.stack = ?stack,
+                    error.trace = %trace,
+                    $($field)*
+                },
+                "{} {}",
+                $message,
+                $error
+            );
+        }
+    };
+    ($error:expr, $message:expr) => {
+        {
+            $crate::trace_error!($error, $message, {})
+        }
+    };
+}
+
+/// Emitt a tracing error event for anyhow error with rich, structured context.
+/// It capturing the error using the experimental Rust [std::error::Report](https://doc.rust-lang.org/stable/std/error/struct.Report.html)
```

Comment:
```suggestion
/// Emit a tracing error event for anyhow error with rich, structured context.
/// It captures the error using the experimental Rust [std::error::Report](https://doc.rust-lang.org/stable/std/error/struct.Report.html)
```

Same typos as above?